### ISSUE TITLE
Damage taken every 10-20 seconds while in sealed room

### ISSUE THREAD

**Issue by Lennywasafish** [Slimefun-Addon-Community/Galactifun#68]
## Description (Required)

I created a small base in the earth's orbit in an enclosed space with iron blocks and glass. I have oxygen sealers and space heaters to stop damage which works for the most part, but every 10 seconds or so, I still seem to take damage. Every 4-5 hits of damage is freezing damage

## Steps to reproduce the Issue (Required)

I have done everything including trying every type of block, making it a smaller space,  removing windows, and removing automatic doors and it still consistently happens.

## Expected behavior (Required)

I would take no damage

## Server Log / Error Report

N/A

## Environment (Required)

 - Minecraft Version: Paper - 259 1.17.1
 - Slimefun Version: DEV - 968
 - Plugin Version: DEV - 5

**Comment by Seggan** [comment]
try checking with `gf sealed <number>`. find the lowest number that says "not sealed"

**Comment by Seggan** [comment]
correction: i meant highest. consider anything above 10000 as that your base is not fully sealed

**Comment by Lennywasafish** [comment]
I ran the command and it returned `/gf sealed 1000000` and it returned `Sealed 1.554 ms`. I am still taking damage every 10 or so seconds though

**Comment by Seggan** [comment]
Find the highest number that says "Not sealed". **This is very important for this issue**

**Comment by Lennywasafish** [comment]
> Find the highest number that says "Not sealed". **This is very important for this issue**

Regardless of the number I put in, it always gives the output of sealed. I have tried up to 1e9 before it gives me an internal error

**Comment by Seggan** [comment]
Try going down. i.e. 1000 or 5000

**Comment by Seggan** [comment]
Start from 100

**Comment by Lennywasafish** [comment]
> Start from 100 then work your way up to 100000

100-1000 gives the output of not sealed but anything above that gives sealed

**Comment by Seggan** [comment]
Hmm

**Comment by Lennywasafish** [comment]
> Hmm

updated reply with exacy numbers

**Comment by Seggan** [comment]
What do your Space Heaters say?

**Comment by Lennywasafish** [comment]
> What do your Space Heaters say?

operational

**Comment by Seggan** [comment]
Edited

**Comment by Lennywasafish** [comment]
I get no output

**Comment by Lennywasafish** [comment]
As stated in my original report, I take damage from lack of oxygen the first 5 times, then I take freezing damage

**Comment by Seggan** [comment]
Yes, ik, I was trying to diagnose further

**Comment by Lennywasafish** [comment]
If you'd like, I can add you on discord and dm you the ip to my server and you can test for yourself

**Comment by Seggan** [comment]
Strange that other have not reported it yet

**Comment by Lennywasafish** [comment]
Yeah it seems like a problem specific to me

**Comment by Seggan** [comment]
> If you'd like, I can add you on discord and dm you the ip to my server and you can test for yourself

I can't unfortunately, my MC acc is being used rn

**Comment by Lennywasafish** [comment]
Ok no worries

**Comment by Lennywasafish** [comment]
I can send a screen recording if you'd like

**Comment by Seggan** [comment]
Sure

**Comment by Lennywasafish** [comment]
[https://imgur.com/LurrPYy](url)

**Comment by Seggan** [comment]
Super fans work only horizontally and diagonally (on the x/z axis)

**Comment by Seggan** [comment]
Try moving them

**Comment by Lennywasafish** [comment]
I've placed them everywhere, no difference, the problem also persists in a 5x5 room without superfans so that isn't the issue

**Comment by Seggan** [comment]
This is *very* strange

**Comment by Seggan** [comment]
Also take note of #69 perhaps that is the problem?

**Comment by Lennywasafish** [comment]
Yeah I made that issue too, that only applies to trapdoors that are on the bottom side of the block

**Comment by Lennywasafish** [comment]
And it happens while standing on iron blocks too

**Comment by Seggan** [comment]
I don't know how to resolve this, I do not know what is causing it. Sorry.

**Comment by Lennywasafish** [comment]
Ok, thanks for trying

**Comment by orangeowie4** [comment]
this happens to me too sometimes

**Comment by qwertyuioplkjhgfd** [comment]
Can you add me on discord? (Char#5513) Maybe I'll give this issue a tackle

**Comment by Ironnoob73** [comment]
I have the same problem, although it wasn't fatal.

**Comment by Seggan** [comment]
Please send me any errors and the output of `/gf effects`

**Comment by JTechGaming** [comment]
i have same problems

**Comment by JTechGaming** [comment]
https://youtu.be/GWwV8FELXTM

**Comment by Derbosik** [comment]
Same issue here

**Comment by GallowsDove** [comment]
The only reason I can think of why this is happening is that in ProtectingBlock 
```
private void uniqueTick() {
    //noinspection deprecation
    Galactifun.protectionManager().clearProtectedBlocks();
    for (BlockPosition l : new HashSet<>(allBlocks)) {
        updateProtections(l);
    }
}
```
The protection blocks are cleared first and only then are they readded. There can be a brief window where the protection isn't added yet, but on another thread, it can be checked whether that protection exists, thus creating a race condition. 
A possible fix could be to calculate all the protections before clearing them and just replace the old ones with the new ones.

I'm not sure about this tho, so let me know what you think

**Comment by Seggan** [comment]
I am aware of the problem but still have no idea how to fix it
